### PR TITLE
[BEAM-975] Improve default connection options, javadoc and style in MongoDbIO

### PR DESCRIPTION
**Pull request by jbonofre**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [X] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [X] Make sure tests pass via `mvn clean verify`.
 - [X] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [X] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by jbonofre** [comment]
R: @dhalperi 
CC: @davorbonaci

File: sdks/java/io/mongodb/src/main/java/org/apache/beam/sdk/io/mongodb/MongoDbIO.java [hunk 1]
```diff
@@ -218,61 +298,64 @@ public void populateDisplayData(DisplayData.Builder builder) {
...
-        desiredBundleSizeBytes = estimatedSizeBytes / spec.numSplits();
-      }
+      try (MongoClient mongoClient = new MongoClient(new MongoClientURI(spec.uri()))) {
+        MongoDatabase mongoDatabase = mongoClient.getDatabase(spec.database());
+
+        List<Document> splitKeys;
+        if (spec.numSplits() > 0) {
+          // the user defines his desired number of splits
+          // calculate the batch size
+          long estimatedSizeBytes = getEstimatedSizeBytes(options);
```
Comment (dhalperi) [inline sdks/java/io/mongodb/src/main/java/org/apache/beam/sdk/io/mongodb/MongoDbIO.java]: do you want a variant where you can pass the already-existing client into the connection? That way you don't have to open two connections here, you can reuse the existing one for estimating size.

**Comment by davorbonaci** [comment]
Any updates?

**Comment by jbonofre** [comment]
I'm updating this pull request.

**Comment by jbonofre** [comment]
@dhalperi @davorbonaci I updated this pull request:
1. rebased on master
2. fix javadoc
3. be able to use an existing Mongo client in `estimatedSizeBytes()`

**Comment by jbonofre** [comment]
Thanks, let me add a couple of tests.

**Comment by jbonofre** [comment]
I'm waiting Jenkins build before merging.

**Pull request closed without merging**